llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -1140,6 +1144,53 @@ void VPWidenRecipe::execute(VPTransformState &State) {
 #endif
 }
 
+void VPWidenEVLRecipe::execute(VPTransformState &State) {
+  assert(State.UF == 1 && "Expected only UF == 1 when vectorizing with "
```

Comment:
Sorry, I don't know much about vectorized design. I would like to know why UF is forced to be set to 1?

Reply:
It's only for EVL-vectorization, as for now it's easier not to handle multiple `get.vector.length`, where one or many could return `0`, thus following instructions will process 0 elements. Another worry I have is we will have to emit extra checks for zero EVL if instruction does ignore it, like RISC-V's `vmv.x.s` if it's generated within a loop